h/streamer/messages.py

```diff
@@ -91,33 +93,48 @@ def handle_annotation_event(message, socket):
     annotation event, otherwise a dict containing information about the event.
     """
     action = message['action']
-    annotation = storage.annotation_from_dict(message['annotation'])
 
     if action == 'read':
         return None
 
     if message['src_client_id'] == socket.client_id:
         return None
 
-    if annotation.get('nipsa') and (
-            socket.request.authenticated_userid != annotation.get('user', '')):
+    # We don't send anything until we have received a filter from the client
+    if socket.filter is None:
         return None
 
-    if not _authorized_to_read(socket.request, annotation):
+    notification = {
+        'type': 'annotation-notification',
+        'options': {'action': action},
+    }
+    id_ = message['annotation_id']
+
+    # Return early when action is delete
+    if action == 'delete':
+        notification['payload'] = [{'id': id_}]
+        return notification
+
+    annotation = storage.fetch_annotation(socket.request.db, id_)
+
+    if annotation is None:
```

Comment:
Yeah, unfortunately that's not true. If I issue an update and a delete in quick succession it's perfectly possible that this function won't process the update until after the annotation ha been deleted.